tasks/http_upload.js

```diff
@@ -59,6 +59,8 @@ module.exports = function(grunt) {
             headers: options.headers,
             multipart: true,
             data: reqData
+          }).on('error',function(e){
+            grunt.fail.warn('Failed Uploading: ' + hostNames[i] + "\n" + e.message); 
```

Comment:
What's `hostNames[i]`?
This gives me a "Fatal error: hostNames is not defined".

This probably would be better:
```
grunt.fail.warn('Failed uploading (error code: ' + e.message + ')'); 
```

WDYT?